File: AvoidanceFlight/Assets/Scripts/GameOver.js
#pragma strict

var GameOverTheme : AudioSource;
var SoundOnce : boolean;

function Start () {
	var AudioSources = GetComponents.<AudioSource>();
	GameOverTheme = AudioSources[0];
	SoundOnce = true;
}

function Update () {
	if (GetComponent.<Text>().enabled == true && SoundOnce) {
		GameOverTheme.Play();
		SoundOnce = false;
	}
}

File: AvoidanceFlight/Assets/Scripts/GameOver.cs
using UnityEngine;
using System.Collections;

[System.Serializable]
public partial class GameOver : MonoBehaviour
{
    public AudioSource GameOverTheme;
    public bool SoundOnce;
    public virtual void Start()
    {
        AudioSource[] AudioSources = this.GetComponents<AudioSource>();
        this.GameOverTheme = AudioSources[0];
        this.SoundOnce = true;
    }

    public virtual void Update()
    {
        if ((this.GetComponent<UnityEngine.UI.Text>().enabled == true) && this.SoundOnce)
        {
            this.GameOverTheme.Play();
            this.SoundOnce = false;
        }
    }

}